metaflow/plugins/kfp/kfp.py

```diff
@@ -406,18 +408,19 @@ def create_workflow_yaml_file(
             sanitize_k8s_name(self.name), max_run_concurrency
         )
 
-        with open(output_path, "a") as yaml_file:
-            yaml_file.write("\n---\n")
-            yaml.safe_dump(config_map, yaml_file, default_flow_style=False)
+        if not KFP_ARGO_COMPILE_WORKFLOW_ONLY:
```

Comment:
As we discussed before, moving business logic (when to create cron workflow) to aip-doit-module or CICD code may be cleaner.

Reply:
We don't create a CronWorkflow when running a Workflow, see 

https://github.com/zillow/metaflow/blob/tz/AIP-7300-argo-cicd/metaflow/plugins/kfp/kfp.py#L356-L386

Reply:
The idea is to:
1) Follow Metaflow Argo Plugin
2) Behaviors about a Flow can all be defined as step and flow decorators.
3) The Flow can be compiled into a single yaml representing the behavior declared in python Flow file.

That is why all related behavior is onefile:

1. The argo workflow
2. The configmap containing the max concurrent runs synchronization semaphore value.
3. The argo CronWorkflow

Reply:
I don't think I want to block this PR further.

Though I can try to explain my preference in more clarity. My ask is mostly a small refactoring for code structure.

I think low level interface (`kfp.py`) should do exactly one thing per function. For example, `netflix/metaflow` has [register_workflow_template](https://github.com/Netflix/metaflow/blob/ec8fd6c4a161b13a0f3765efb1e7fea92cdc2baa/metaflow/plugins/argo/argo_client.py#L55) and [schedule_workflow_template](https://github.com/Netflix/metaflow/blob/master/metaflow/plugins/argo/argo_client.py#L270) separately.

At layers where user scenario is clear (`kfp_cli.py` or `ci-cd-template` corresponding to notebook or CI pipeline deployment), we can then have functions combining multiple lower level interface, such as deploying (WorkflowTemplate) and scheduling (CronWorkflow) the workflow in one call. For example, `netflix/metaflow` combines flow deploy and trigger call in one [create](https://github.com/Netflix/metaflow/blob/ec8fd6c4a161b13a0f3765efb1e7fea92cdc2baa/metaflow/plugins/argo/argo_workflows_cli.py#L254) function in `argo_workflow_cli`.

Reply:
will handle in this PR https://github.com/zillow/metaflow/pull/240
with ticket [AIP-7535](https://zbrt.atl.zillow.net/browse/AIP-7535) WFSDK: Separate register and schedule workflow functions